modules/core/plan/plan.module

```diff
@@ -26,6 +27,18 @@ function plan_help($route_name, RouteMatchInterface $route_match) {
   return $output;
 }
 
+/**
+ * Implements hook_ENTITY_TYPE_delete().
+ */
+function plan_plan_delete(EntityInterface $entity) {
+
+  // Delete all plan_record entities associated with the plan.
+  $plan_records = \Drupal::entityTypeManager()->getStorage('plan_record')->loadByProperties(['plan' => $entity->id()]);
+  foreach ($plan_records as $plan_record) {
+    $plan_record->delete();
+  }
+}
+
```

Comment:
This LGTM. The only issue I can think of is if a plan had many many plan record entities associated with it, this might crash when loading all of them? I'm sure if we realistically need to worry about this though. We could probably do better by loading just the plan_record entity IDs into an array, and then use each ID to load and delete entity objects one at a time

Reminds me of this issue https://www.drupal.org/project/drupal/issues/2577417

Reply:
I had a similar thought @paul121. I wondered about using Batch API but I'm not sure that's possible in `hook_ENTITY_TYPE_delete()`. I like your suggestion of loading them one at a time. I've pushed a commit that refactors the hook to do that (loading IDs via an entity query first).

Tests are still passing. Re-requesting your review...